packages/glimmer-runtime/tests/ember-component-test.ts

```diff
@@ -2042,6 +2049,23 @@ QUnit.test('Curly component hooks (attrs as self props)', function() {
   assertFired(instance, 'didRender', 3);
 });
 
+QUnit.test('statements have proper moduleName assigned', function() {
+  expect(3);
+  class MyComponent extends EmberishGlimmerComponent {}
+
+  let original = env.statement;
+  env.statement = function(statement: StatementSyntax, symbolTable: SymbolTable) {
+    if (statement instanceof Text) {
+      equal(statement.content, symbolTable.getModuleName(), 'moduleName is correct');
+    }
+    return original.call(env, statement, symbolTable);
+  }
+
+  env.registerEmberishGlimmerComponent('my-component', inspectHooks(MyComponent), 'my-component{{yield}}');
+
+  appendViewFor('<div>-top-level{{#my-component}}-top-level{{/my-component}}</div>');
```

Comment:
the point of asking you to do this was to check the getModuleName() symbol in the top level, layout and block context.